### ISSUE TITLE
GMT-7 sync: broken files for all photos

### ISSUE THREAD

**Issue by lbestock** [arch-kiosk/arch-kiosk-office#1184]
PVD 0.13, recording 10.6.3. I downloaded and worked in two workstations today, uploaded and synchronized. All of the images I made on the iPad were unknown at synchronization and resulted in broken files. They really look just fine on the ipad. Synchronization alerted me that this would be a problem: 
![image](https://user-images.githubusercontent.com/49291909/126720841-3b97e818-f778-495e-8530-9630aaba3f03.png)

(I mean, you can look at the log, too. Uh, timezone. This was at about 4:15pm California time on July 22, so 1:15am July 23 in Germany.)
I am very seriously not at my best and so it's possible I screwed things up. I certainly made errors in the process, but I think they should not have this result. 

Process, including multiple anomalies that may figure somewhere:
I uploaded the ipad first. I first told it to synchronize without importing the upload, which of course failed. Might that step have wrecked stuff? Then I uploaded it properly.
My mac I downloaded and worked in only after I had uploaded and imported the ipad, but I did not try to synchronize again because I knew I would have another workstation. I screwed up uploading my mac workstation multiple times, first working in an old file which of course wouldn't import. Then I worked in the proper download but had to try to upload three times before I could (that part being crappy internet in the living room where I am working while being ignored by all three kittens). 

So it definitely wasn't a smooth process, but if one of those things caused all the files to be broken then we still need to find a way to not have it happen. And I can't really imagine how screwing up with the mac workstation would lead to broken files on the records from the ipad, I just report in full.

**Comment by lbestock** [comment]
The broken images included those made with the camera of the iPad and the signature tool.

**Comment by lbestock** [comment]
Oh wait. Sick brain. ALL the images I put in today were broken, including those I added from my mac workstation.

**Comment by urapadmin** [comment]
>!Synchronization collision in table "constants", field "value", record "75fb9997-ebe7-1345-926f-a7694c4c22bb": 
>[5636/6300: synchronization.INFO at 2021-07-22 19:16:48,029]: |  id  ||  value  ||  value_ts  ||  category  ||  comment  ||  field_type  ||  sync  ||  uid  ||  created  ||  modified  ||  modified_by  ||  repl_deleted  ||  repl_tag  ||  repl_workstation_id  |
WINS: >id_device, Laurel's 13-inch, None, system, None, None, 1, 75fb9997-ebe7-1345-926f-a7694c4c22bb, 2020-04-23 01:25:31, 2021-07-22 15:31:09, LDB, False, 0, ldb_mac
id_device, Flinders, None, system, None, None, 1, 75fb9997-ebe7-1345-926f-a7694c4c22bb, 2020-04-23 01:25:31, 2021-07-22 12:28:27, LDB, False, 0, ldb_ipad

[MCP_5636_kiosk_Thu_220721-1916.log](https://github.com/urapadmin/kiosk/files/6872265/MCP_5636_kiosk_Thu_220721-1916.log)

**Comment by urapadmin** [comment]
I have a hunch. This system is an offline recording system not designed to be used in scenarios where server and client are in different time-zones. And here we might have one of the side effects: California is 3 hours back - not our usual direction. 
- The server notes 22.07.2021 15:11:25 as fork time for workstation ldb_ipad.
- The utc-time was set to -07:00:00 - Laurel has set the filemaker database manually to local time - the workstation is configured with default GMT (it would not have helped to change that)
- all images where changed/added between 22.VII. 12:32 and 13:33, so BEFORE the fork time. 
- the database does not export any images prior to fork time because it thinks that those cannot have been changed since download from server - which is correct, but cannot deal with timezones!
-

**Comment by urapadmin** [comment]
in principle I do not count this as an error. It is an invalid use case for v1. 
- We are just lucky that it does not happen in Swansea because everything East of Providence is always ahead of the server and then the issue does not occur - modifications are always later than the fork time. Going west teams must set their devices to the server's timezone.
- To support the single issue I could store the server's GMT in the workstation database. That way the workstation can calculate a fork time for the local time zone and compare the image dates with that. Images later than the local fork time would be exported. 
- BUT: In the end this should screw up synchronization as well, shouldn't it? If I change a single record, unit or dayplan or whatever. That would be a change in the past as well and would it not be ignored by synchronization?

**Comment by urapadmin** [comment]
see #293

**Comment by urapadmin** [comment]
[another unopened ldb_mac import - MCP_7260_kiosk_Thu_220721-1850.log](https://github.com/urapadmin/kiosk/files/6872324/another.unopened.ldb_mac.import.-.MCP_7260_kiosk_Thu_220721-1850.log)
[cancelled import - ldb_mac - MCP_7324_kiosk_Thu_220721-1835.log](https://github.com/urapadmin/kiosk/files/6872325/cancelled.import.-.ldb_mac.-.MCP_7324_kiosk_Thu_220721-1835.log)
[forknexport - ldb_mac - kiosk_Thu_220721-1801.log](https://github.com/urapadmin/kiosk/files/6872326/forknexport.-.ldb_mac.-.kiosk_Thu_220721-1801.log)
[import ldb_ipad - _kiosk_Thu_220721-1801.log](https://github.com/urapadmin/kiosk/files/6872327/import.ldb_ipad.-._kiosk_Thu_220721-1801.log)
[import ldb_mac -  MCP_1280_kiosk_Thu_220721-1913.log](https://github.com/urapadmin/kiosk/files/6872328/import.ldb_mac.-.MCP_1280_kiosk_Thu_220721-1913.log)
[import ldb_mac cancelled - kiosk_Thu_220721-1856.log](https://github.com/urapadmin/kiosk/files/6872329/import.ldb_mac.cancelled.-.kiosk_Thu_220721-1856.log)
[import unopened ldb_mac import - kiosk_Thu_220721-1830.log](https://github.com/urapadmin/kiosk/files/6872330/import.unopened.ldb_mac.import.-.kiosk_Thu_220721-1830.log)
[in vain synchronization - kiosk_Thu_220721-1800.log](https://github.com/urapadmin/kiosk/files/6872331/in.vain.synchronization.-.kiosk_Thu_220721-1800.log)
[synchronization - MCP_5636_kiosk_Thu_220721-1916.log](https://github.com/urapadmin/kiosk/files/6872332/synchronization.-.MCP_5636_kiosk_Thu_220721-1916.log)

**Comment by urapadmin** [comment]
@lbestock :
since you are in the West, please prepare your workstation afresh and 
- add a single record somewhere. Let's say a narrative. 
- change a different narrative
- Don't change anything else. 
- import and synchronize this one workstation. Make sure no other workstations slips into this synchronization run.
- report the synchronization time here, so I can find the logs.
- prepare your workstation again and look up if the changes are there.

**Comment by lbestock** [comment]
CA - one narrative added, one changed.
No other workstations than ldbmac snuck in.
Synchronized 8:25am California time (only half a cup of tea - can't convert to GMT yet).
No reported problems from the little synch synopsis.
I haven't downloaded my workstation again but no need - director's view tells me the change to one narrative and the new narrative are both there.

**Comment by urapadmin** [comment]
so it is only an issue for files. That would make it at perhaps worth the while to find a solution for it. In the end only a real timezone-aware design can handle this safely. v1 is a local field recording system.

**Comment by lbestock** [comment]
For fun I downloaded JIAAWmac, which had been forked ages ago, and added a handful of images, and synchronized at 3:15pm CA time. I still got a scary “there is no file matching” report three times. And yet, in the file repository are all five of the images I just added.

**Comment by urapadmin** [comment]
> 
> 
> For fun I downloaded JIAAWmac, which had been forked ages ago, and added a handful of images, and synchronized at 3:15pm CA time. I still got a scary “there is no file matching” report three times. And yet, in the file repository are all five of the images I just added.

this is different and referenced now in #1188

**Comment by urapadmin** [comment]
at least I should consider an additional safety feature that alerts users in filemaker to the fact that they are ahead of the server, timezone-wise. The server would still have to register its timezone with filemaker but it is less work to just produce an alert.

**Comment by urapadmin** [comment]
the server's utc offset is now defined as "utc_offset_server" within filemaker.

**Comment by urapadmin** [comment]
### @test 10.8
- [ ] when having the local device set to a timezone that is earlier (West) of the server's timezone you should get a message when starting that alerts you to the fact but allows you to proceed. Don't trick yourself, if your local timezone differs from what the server told your device to be in (defined in the field manager for the workstation) you will get the error message about that one first and never get to the warning about an earlier timezone.

**Comment by lbestock** [comment]
Totally works. That's pretty cool.

**Comment by urapadmin** [comment]
### recording 15.1.4
- because I reopened #2075 I have to reopen this, too. 
- Please give this a full testing again

**Comment by urapadmin** [comment]
that's fine. For the time being khpp got the help they needed and are back in the game.

**Comment by luizaogs** [comment]
Oh, it's a recording update. Nevermind.

**Comment by urapadmin** [comment]
That is super weird. We use webp at this point for thumbnails in the browser. But it should be jpgs when transported to the recording system. This is not related to the original issue, so lets open up a new ticket.

**Comment by luizaogs** [comment]
Ok. The rest here was fine, so I close it.
